timesketch/frontend/src/components/Sketch/TimeFilter.vue

```diff
@@ -16,113 +16,219 @@ limitations under the License.
+                <option v-for="option in intervals" :value="option.value" :key="option.value">{{ option.text }}</option>
+              </select>
+            </span>
           </p>
         </div>
       </div>
     </div>
 
+    <!-- Time range -->
     <div class="field is-horizontal">
-        <div class="field is-grouped">
+      <div class="field-body" style="display:flex; align-items:center;">
+
+        <b-radio v-model="radio" native-value="range"></b-radio>
+        <div class="field">
           <p class="control">
-            <a :disabled="!startDateTime" class="button is-light" v-on:click="formatDateTime()">
-              <span class="icon is-small">
-                <i class="fas fa-magic"></i>
-              </span>
-              <span>Format</span>
-            </a>
+            <input class="input" ref="startInput" v-model.trim="startDateTime" type="text" :disabled="!isSelected('range')" :placeholder="getPlaceholder('range')"
+            v-on:change="startDateTime && formatDateTime()" v-on:keyup.enter="endDateTime ? submit() : jumpTo('endInput')">
           </p>
+        </div>
+
+        <span style="margin-right:10px;">&rarr;</span>
+
+        <div class="field">
           <p class="control">
-            <button v-if="selectedChip" :disabled="!(startDateTime && endDateTime)" class="button is-success is-outlined" v-on:click="update">Update</button>
-            <button v-else :disabled="!(startDateTime && endDateTime)" class="button is-success is-outlined" v-on:click="submit">+ Add time range</button>
+            <input class="input" ref="endInput" v-model.trim="endDateTime" type="text" :disabled="!isSelected('range')" :placeholder="getPlaceholder('range')"
+            v-on:change="endDateTime && formatDateTime()" v-on:keyup.enter="submit()">
           </p>
         </div>
+      </div>
+    </div>
+
+    <!-- Submit button -->
+    <div class="field is-horizontal">
+      <div class="field is-grouped">
+        <p class="control">
+          <button :disabled="!hasAllInputs()" class="button is-success" v-on:click="submit">{{ selectedChip ? 'Update' : 'Create' }}</button>
+        </p>
+      </div>
     </div>
 
   </div>
 </template>
 
 <script>
 export default {
-  props: ['start', 'end', 'selectedChip'],
+  props: ['selectedChip'],
   data () {
     return {
-      startDateTime: this.start,
-      endDateTime: this.end,
-      chip: this.selectedChip
+      startDateTime: '',
+      endDateTime: '',
+      chip: this.selectedChip,
+      radio: 'interval',
+      offsetStart: '',
+      offsetMinus: 5,
+      offsetPlus: 5,
+      intervals: [
+        {'text': 'Second', 'value' : 's'},
+        {'text': 'Minute', 'value' : 'm'},
+        {'text': 'Hour', 'value': 'h'},
+        {'text': 'Day', 'value': 'd'},
+      ],
+      selectedInterval: 'm',
+    }
+  },
+  created: function () {
+    if (!this.chip) {
```

Comment:
You can also mark the prop (selectedChip) as required.

Something like:
```
 props: {
   selectedChip: {
     type: Object,
     required: true
   }
```

Reply:
I don't think that's necessary because we treat it as optional and only use "selectedChip" to be able to update existing time filters. That's our way to distinguish if we're dealing with a new chip or if we're updating the existing one.